operations/configuration_management/Write_Ansible_Code_for_CFS.md

```diff
@@ -38,15 +38,67 @@ CFS will handle scaling up Ansible to run on many hosts, but there are still pla
 ### Using image customization
 
 * Use image customization when possible; doing so will improve boot times by limiting how many times a task is run. Configuration that is the same for all nodes of the same type will benefit from image customization.
-* Use different playbooks for image customization and node personalization.
-Moving image customizations tasks to their own playbook can remove the need to evaluate conditionals in a shared playbook, as well as ensuring that tasks are not accidentally running in both modes needlessly.
+* Use the `cfs_image` host group to distinguish between image customization and node personalization.
+  This allows image customization to be identified in the `hosts` parameter, removing the need to evaluate conditionals, and ensuring that tasks are not accidentally running in both modes needlessly.
 * See [Target Ansible Tasks for Image Customization](Target_Ansible_Tasks_for_Image_Customization.md) for more information on writing for image customization.
 
 ### Disable fact gathering
 
 * Turn off facts that are not needed in a playbook by setting `gather_facts: false`. If only a few facts are required, it is also possible to limit fact gathering by setting `gather_subset`.
 For more information on `gather_subset`, see the external [Ansible module setup](https://docs.ansible.com/ansible/latest/modules/setup_module.html) documentation.
-* Avoid importing playbooks in other playbooks.  This will trigger fact gathering for each imported playbook, potentially collecting the same information multiple times.
+* Reducing fact gathering time is especially important when importing multiple playbooks from a top level playbook.
+  Fact gathering will trigger for each imported playbook, potentially collecting the same information multiple times.
+  
+### `group_by` and `add_host`
+
+The `group_by` and `add_host` modules can both be used to dynamically generate new hosts groups for the Ansible inventory.  This is useful when for hosts that be grouped according to a common property, and when plays or large sections of plays are intended to only target a particular kind of host.  Examples of this might include grouping by operating system, hardware type, or a hardware property such as the presence of a GPU.
```

Comment:
Second sentence proposed replacement:
These modules proved useful when hosts can be group according to a common property. Then, plays can be designed to only target that particular group.